.github/workflows/build.yml

```diff
@@ -0,0 +1,63 @@
+name: Build
+on:
+  pull_request: {}
+  push:
+    branches:
+      - master
+    tags:
+      - v*
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  test:
+    name: Test (${{ matrix.scenario }})
+    runs-on: ubuntu-latest
+    env:
+      CI: 'true'
+    strategy:
+      matrix:
+        scenario:
+          - default
+          - default-with-lockfile
+          - default-with-jquery
+          - default-with-fastboot
+          - release
+          - beta
+          - canary
+          - lts-3.16
+          - lts-3.12
+          - lts-3.8
+          - lts-3.4
+          - lts-2.18
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+      - name: Set up Volta
+        uses: volta-cli/action@v1
+      - name: Install dependencies (yarn)
+        if: matrix.scenario == 'default-with-lockfile'
+        run: yarn install --non-interactive --frozen-lockfile
+      - name: Install dependencies (yarn)
+        if: matrix.scenario != 'default-with-lockfile'
+        run: yarn install --non-interactive --no-lockfile
+      - name: Setup ember-try scenario
+        if: matrix.scenario != 'default' && matrix.scenario != 'default-with-lockfile'
+        run: >-
+          yarn ember try:one ember-${{ matrix.scenario }} --skip-cleanup ---
+          '[ -z "$EMBER_OPTIONAL_FEATURES" ] || echo "::set-env name=EMBER_OPTIONAL_FEATURES::$EMBER_OPTIONAL_FEATURES"'
+      - name: Lint (hbs)
+        run: yarn lint:hbs
+      - name: Lint (js)
+        run: yarn lint:js
+      - name: Build
+        if: matrix.scenario != 'default-with-fastboot'
+        run: yarn ember build --environment test
+      - name: Run test
+        if: matrix.scenario == 'default-with-fastboot'
+        run: yarn test:fastboot
+      - name: Run test
+        if: matrix.scenario != 'default-with-fastboot'
+        # Due to a bug in ember-cli, running `ember test` with `--path` doesn't set `EMBER_ENV=test`
+        # See https://github.com/ember-cli/ember-cli/issues/8922
+        run: EMBER_ENV=test yarn test --path dist
```

Comment:
Wouldn't that run `ember test` twice for each scenario, once with the ember-try scenario's dependencies (L47), and once with the default ones here? Also linting for each scenario.

I tried to replicate the default Travis setup as much as possible, have one job that runs linting and default tests, one with floating deps, and then the matrix of ember-try scenarios. E.g. like this: https://github.com/kaliber5/ember-in-element-polyfill/blob/master/.github/workflows/ci.yml

Reply:
L47 only does the `yarn install` and sets the optional feature ENV variable, because I passed a command to ember try instead of the default (ember test). I like to break down the steps as much as possible so it makes it easier to see what failed and find the logs. 

As for floating dependencies, all of the scenarios are floating by default (`—no-lockfile`) and I added one that tests with lockfile specifically. This actually confused me a little because the default Travis yaml also uses `—no-lockfile` by default so I don’t actually understand why the floating dependencies one is needed at all.